### PR TITLE
move ingress to networking v1

### LINKED ISSUE #188: Move ingress to v1 version
- ```W0517 04:43:20.660005       1 warnings.go:70] networking.k8s.io/v1beta1 Ingress is deprecated in v1.19+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress```

### PR DESCRIPTION
**Pull request by AdheipSingh**
Fixes #188

### Description

<hr>

This PR has:
- [x] been tested on a real K8S cluster to ensure creation of a brand new Druid cluster works.
- [x] been tested for backward compatibility on a real K*S cluster by applying the changes introduced here on an existing Druid cluster. If there are any backward incompatible changes then they have been noted in the PR description.
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added documentation for new or modified features or behaviors.

<hr>

##### Key changed/added files in this PR
 * `types.go`
 * `handler.go`
 * `crds/`
 * `Makefile`

### CODE REVIEW

**Comment by AdheipSingh** [comment]
so the build is failing due to kubebuilder smoke tests provisioning a cluster not supporting networking v1. 
Need to update kubebuilder to v3.

**Comment by AdheipSingh** [comment]
kubebuilder v3 isnt starting with make tests. Mostly an issue in controller-runtime envtest.
marking this PR as draft, will take the envtest issue in another PR. 

## Error log
```
{"level":"error","ts":1622389786.2629309,"logger":"controller-runtime.test-env","msg":"unable to start the controlplane","tries":4,"error":"timeout waiting for process kube-apiserver to start","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/home/adheip/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132\nsigs.k8s.io/controller-runtime/pkg/log.(*DelegatingLogger).Error\n\t/home/adheip/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/log/deleg.go:144\nsigs.k8s.io/controller-runtime/pkg/envtest.(*Environment).startControlPlane\n\t/home/adheip/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/envtest/server.go:291\nsigs.k8s.io/controller-runtime/pkg/envtest.(*Environment).Start\n\t/home/adheip/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/envtest/server.go:253\ngithub.com/druid-io/druid-operator/controllers/druid.setupK8Evn\n\t/home/adheip/rill/druid-operator/controllers/druid/suite_test.go:40\ngithub.com/druid-io/druid-operator/controllers/druid.TestAPIs\n\t/home/adheip/rill/druid-operator/controllers/druid/suite_test.go:53\ntesting.tRunner\n\t/home/adheip/.go/src/testing/testing.go:992"}
--- FAIL: TestAPIs (101.07s)
    suite_test.go:41: 
                Error Trace:    suite_test.go:41
                                                        suite_test.go:53
                Error:          Received unexpected error:
                                failed to start the controlplane. retried 5 times: timeout waiting for process kube-apiserver to start
                Test:           TestAPIs
FAIL
coverage: 18.4% of statements
FAIL    github.com/druid-io/druid-operator/controllers/druid    101.094s
?       github.com/druid-io/druid-operator/controllers/druid/ext        [no test files]
FAIL
Makefile:18: recipe for target 'test' failed
make: *** [test] Error 1
```

**Comment by AdheipSingh** [comment]
i am able to finally run tests successfully locally, the kube apiserver binary needs to be updated. have updated in the Dockerfile.
@himanshug in case you want to run locally the tests, update the kube-apiserver cli to 1.19 version. 
This PR is ready for review. :)

**Comment by AdheipSingh** [comment]
BTW just to let you know i tried to use kubebuilder v3.0.1 but it does not have a bin folder with all the kubebuilder tools.
Plus we arnt updating the go.mod using kubebuilder so it hardly matters post scaffold :)

File: Makefile [hunk 1]
```diff
@@ -43,6 +43,7 @@ deploy: manifests
 # Generate manifests e.g. CRD, RBAC etc.
 manifests: controller-gen
 	$(CONTROLLER_GEN) $(CRD_OPTIONS) paths="./..." output:crd:artifacts:config=deploy/crds/
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) paths="./..." output:crd:artifacts:config=chart/templates/crds/
```
Comment (AdheipSingh) [inline Makefile:46]: I have added this too make sure, both helm folder crd and deploy folder crds are updated at once.

**Review by nishantmonu51 — APPROVED** [review]
👍

**Pull request merged**